lib/include/openturns/OTtestcode.hxx

```diff
@@ -454,6 +429,596 @@ void assert_equal(const T & a, const T & b, const String errMsg = "")
+    {
+      const Sample sample(distribution_.getSample(5));
+      for (UnsignedInteger i = 0; i < sample.getSize(); ++ i)
+      {
+        const Point x(sample[i]);
+        const Scalar pdf1 = distribution_.computePDF(x);
+        LOGTRACE(OSS() << "pdf    =" << pdf1);
+        const Scalar cdf1 = distribution_.computeCDF(Point({x[0] + epsilon[0]}));
+        const Scalar cdf2 = distribution_.computeCDF(Point({x[0] - epsilon[0]}));
+        const Scalar pdf2 = (cdf1 - cdf2) / (2.0 * epsilon[0]);
+        LOGTRACE(OSS() << "pdf(fd)=" << pdf2);
+        assert_almost_equal(pdf1, pdf2, 1e-3, 1e-3,  "cdf " + distribution_.__repr__());
+      }
+    }
+  }
+
+  void checkDDF() const
+  {
+    if (distribution_.isContinuous())
+    {
+      LOGTRACE(OSS() << "checking DDF...");
+      const Sample sample(distribution_.getSample(10));
+      const Scalar epsilon = std::pow(ResourceMap::GetAsScalar("DistFunc-Precision"), 1.0 / 3.0);
+      for (UnsignedInteger i = 0; i < sample.getSize(); ++ i)
+      {
+        const Point x(sample[i]);
+        const Point ddf1 = distribution_.computeDDF(x);
+        LOGTRACE(OSS() << "ddf    =" << ddf1 << " x=" << x);
+        Point ddf2(distribution_.getDimension());
+        if (ddf1.getDimension() != distribution_.getDimension())
+          throw TestFailed(OSS() << "ddf(x) failed for " << distribution_);
+        for (UnsignedInteger j = 0; j < distribution_.getDimension(); ++ j)
+        {
+          Point xp(x);
+          xp[j] += epsilon;
+          const Scalar pdf1 = distribution_.computePDF(xp);
+          Point xm(x);
+          xm[j] -= epsilon;
+          const Scalar pdf2 = distribution_.computePDF(xm);
+          ddf2[j] = (pdf1 - pdf2) / (2.0 * epsilon);
+        }
+        LOGTRACE(OSS() << "ddf(fd)=" << ddf2);
+        assert_almost_equal(ddf1, ddf2, 1e-3, 1e-3, "ddf " + distribution_.__repr__());
+      }
+    }
+  }
+  
+  void checkCCDF() const
+  {
+    if (distribution_.isContinuous() && !distribution_.isCopula() && (distribution_.getDimension() < 4))
+    {
+      LOGTRACE(OSS() << "checking CCDF...");
+      const Sample sample(distribution_.getSample(5));
+      for (UnsignedInteger i = 0; i < sample.getSize(); ++ i)
+      {
+        const Point x(sample[i]);
+        const Scalar ccdf2 = 1.0 - distribution_.computeCDF(x);
+        LOGTRACE(OSS() << " 1-cdf=" << ccdf2);
+        const Scalar ccdf1 = distribution_.computeComplementaryCDF(x);
+        LOGTRACE(OSS() << "  ccdf=" << ccdf1);
+        assert_almost_equal(ccdf1, ccdf2, 1e-5, 1e-5, "ccdf " + distribution_.__repr__());
+      }
+    }
+  }
+  
+  void checkSurvival() const
+  {
+    if (distribution_.getDimension() == 1)
+    {
+      LOGTRACE(OSS() << "checking survival...");
+      const Point x(distribution_.getRealization());
+      const Scalar survival = distribution_.computeSurvivalFunction(x);
+      LOGTRACE(OSS() << "survival=" << survival);
+      const Scalar survival2 = 1.0 - distribution_.computeCDF(x);
```

Comment:
`computeComplementaryCDF` is exactly for this purpose for a 1-d distribution: please use it here.

Reply:
thats already tested in compute checkCCDF